ymer-test-junit4/src/main/java/com/avanza/ymer/YmerConverterTestBase.java

```diff
@@ -112,22 +115,38 @@ public void testFailsIfCollectionOrMapPropertyOfTestSubjectIsEmpty() throws Exce
     			+ "add at least one element to ensure proper test coverage.: " + emptyFields, emptyFields.isEmpty());
     }
 
+	@Test
+	public void shouldRequireAnnotationsOnCustomConverters() {
+		for (Converter<?, ?> customConverter : getMirroredObjectsConfiguration().getCustomConverters()) {
+			boolean hasReadingConverterAnnotation = customConverter.getClass().getAnnotation(ReadingConverter.class) != null;
+			boolean hasWritingConverterAnnotation = customConverter.getClass().getAnnotation(WritingConverter.class) != null;
+			if (!hasReadingConverterAnnotation && !hasWritingConverterAnnotation) {
+				fail("Custom converter=[" + customConverter.getClass() + "] should be annotated with either @ReadingConverter or @WritingConverter");
+			}
+		}
+	}
+
 	private MirroredObjectTestHelper getMirroredObjectHelper(Class<?> objectClass) {
 		return MirroredObjectTestHelper.fromDefinitions(getMirroredObjectDefinitions(), objectClass);
 	}
 
-	protected abstract Collection<MirroredObjectDefinition<?>> getMirroredObjectDefinitions();
+	protected Collection<MirroredObjectDefinition<?>> getMirroredObjectDefinitions() {
+		return getMirroredObjectsConfiguration().getMirroredObjectDefinitions();
+	}
+
+	protected CustomConversions getCustomConversions() {
+		return new MongoCustomConversions(
+				getMirroredObjectsConfiguration().getCustomConverters()
+		);
+	}
 
-	protected abstract CustomConversions getCustomConversions();
+	protected abstract MirroredObjectsConfiguration getMirroredObjectsConfiguration();
```

Comment:
Warning: This BREAKS compatibility with existing unit tests that extend this class. But the compatibility with `2.x.x` was already broken in 4fcbb39eb267140f60347e0c1349d932dbe43dac .